test/controllers/helpers_test.rb

```diff
@@ -270,6 +273,7 @@ def setup
     @controller.session[:user_return_to] = "/foo.bar"
     @mock_warden.expects(:user).with(:user).returns(nil)
     @mock_warden.expects(:set_user).with(user, scope: :user).returns(true)
+    @mock_warden.expects(:env).returns({})
```

Comment:
Can we add a new test that checks that the `#update_tracked_fields!` method was called on `resource`?